### ISSUE TITLE
clarity on index types

### ISSUE THREAD

**Issue by noahehall** [arangodb/arangojs#648]
arango 3.4.9
rocksdb 5.16.0

[according to the docs](https://www.arangodb.com/docs/stable/indexing-index-basics.html#persistent-index)
[and here](https://www.arangodb.com/docs/stable/indexing-index-basics.html#persistent-index)
- *The persistent index type is deprecated for the MMFiles storage engine. Use the RocksDB storage engine instead, where all indexes are persistent.*
- *index types hash, skiplist and persistent are equivalent when using the RocksDB storage engine.*

what do you mean by equivalent? are they identical? or simply are all persistent i.e. persisted to disk?
some index types are persistent, and then there is the persistent index type...

additionally, my web interface offers the following index types
- geo
- hash
- persistent
- full-text
- skiplist

that one sentence about equivalency is fkin up my ability to choose an index type, which is extremely funny, as I understand which index type im suppose to choose for my specific use case, but to think a sorted index is the equiavelent to hash index, if the docs are correct, is making me question my degree from wikipedia

**Comment by noahehall** [comment]
I just upgraded to 3.6.1
and i now see the equivalency

- skip, hash -> persistent

**Comment by noahehall** [comment]
for any one else arriving, [here are the tests for ensureindex](https://github.com/arangodb/arangodb/blob/master/tests/js/common/shell/shell-index-ensure2-mmfiles.js)

**Comment by jsteemann** [comment]
In the past, we only had the MMFiles storage engine. Here the index types hash, skiplist and persistent are fundamentally different, and the names correctly indicate the type of each index.

Later, we added the RocksDB storage engine to ArangoDB. In order to ensure maximum compatibility and make engine switching as easy as possible, we stuck with the existing index type names even for RocksDB. But, as the documentation states, these index types are identical for the RocksDB storage engine. They have different names but use the same internal implementation.

In order to make things a bit less confusing, we simplified the web interface in 3.6 so that it only offers one of these index types when creating a new index with the RocksDB engine. We picked "persistent" here, because that most clearly indicates the type of index. "hash" and "skiplist" don't really fit for the RocksDB-based indexes, as they are neither of these two things. In RocksDB, all keys are stored in sorted order, so functionally it provides the same behavior and guarantees as a skiplist. But underneath it is using whatever RocksDB is employing (ironically, in-memory data is stored in RocksDB in a skiplist-based structure - but on disk data is stored differently). 

It is still possible to use the APIs to create indexes of types hash, skiplist and persistent. Though internally their characteristics will be identical, and the web interface does not offer the former two types.
I hope this clarifies the confusion at least a bit!
